nixos/modules/hardware/onlykey.nix

```diff
@@ -0,0 +1,73 @@
+{ config, lib, ... }:
+
+with lib;
+
+{
+
+  ####### interface
+
+  options = {
+
+    hardware.onlykey = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enable OnlyKey device (https://crp.to/p/) support.
+        '';
+      };
+    };
+
+  };
+
+  ## As per OnlyKey's documentation piece (hhttps://docs.google.com/document/d/1Go_Rs218fKUx-j_JKhddbSVTqY6P0vQO831t2MKCJC8),
+  ## it is important to add udev rule for OnlyKey for it to work on Linux
+
+  ####### implementation
+
+  config = mkIf config.hardware.onlykey.enable {
+    services.udev.extraRules =
+      ''
+      # UDEV Rules for Teensy boards, http://www.pjrc.com/teensy/
```

Comment:
This whole block appears to be copied from a file. Either:

1. download the file to the containing directory. Use builtins.readFile.
2. remove the superfluous comments about how to install the file. Not relevant.